### ISSUE TITLE
Support SavedModel in WarmStartSettings initialization

### ISSUE THREAD

**Issue by dwyatte** [tensorflow/tensorflow#25400]
Any reason not to support SavedModel in WarmStartSettings initialization? Currently only checkpoints are supported.

Estimator's `warm_start_from` argument accepts a SavedModel, so it would be great to have parity with that for cases where a user needs finer-grained control via WarmStartSettings (e.g., only initializing a subset of variables for transfer learning/fine tuning).

**Comment by allenlavoie** [comment]
Sounds reasonable to me. Although it should already be possible to point warm start to the `variables/` directory of the SavedModel, no? That's just a regular training checkpoint.

**Comment by dwyatte** [comment]
Digging in, Estimators support SavedModels using the logic @allenlavoie mentions, so it is possible to just point WarmStartSettings at `[saved_model_dir]/variables/variables` (note the extra `variables`) (https://github.com/tensorflow/estimator/blob/master/tensorflow_estimator/python/estimator/estimator.py#L2285).

WarmStartSettings makes an explicit call to checkpoint_utils.init_from_checkpoint (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/training/warm_starting_util.py#L364), so it seems like the above logic could replace that.

**Comment by dwyatte** [comment]
I opened a small PR https://github.com/tensorflow/estimator/pull/53/files fixing the issue where WarmStartSettings needed to be initialized from `[saved_model_dir]/variables/variables` instead of `[saved_model_dir]/variables` like @allenlavoie suggested. I suspect this may be for v1 compatibility only at this point, but wanted to put it out there in the interest of closing this issue.

**Comment by sachinprasadhs** [comment]
Hi, Estimator will be deprecated from Tensorflow 2.12 since it uses Tensorflow V1 behavior.
Could you please migrate from Estimator to Keras API which will enable you to perform the aforementioned tasks, such as [model building](https://www.tensorflow.org/guide/keras/custom_layers_and_models), gradient application, [training](https://www.tensorflow.org/guide/keras/customizing_what_happens_in_fit), evaluation, and prediction.
Refer Migration guide here https://www.tensorflow.org/guide/migrate/migrating_estimator for more details. Thanks!

**Comment by dwyatte** [comment]
Agree, we can close this issue due to the deprecation. Thanks for the reminder!
